### DISCUSSION TITLE
Workflow with digiKam

### DISCUSSION THREAD

**Discussion by RobWW** [immich-app/immich#10089] [Q&A]
I am trying to figure out a photography workflow.  I have read  numerous comments that the combination of digiKam and immich works well, with digiKam being used to sort and edit images, and immich being used to display the final product.

However, I am unclear how people integrate the two programs as, from what I can tell, immich does not recognize metadata created by digiKam.  

Does creating a script that reads the digiKam metadata and only adds selected images to immich work well?  Or is the digiKam Export feature used to create a copy of selected files for import into immich?  In either case, I would be concerned about keeping the two programs in sync.  Or am I missing something obvious?

If it helps, I would like to create albums in immich that contain a subset of the total images (potentially based on digiKam labels, such as Pick -> Accepted item).  If the image has been edited, I would like to only add the final version.  Album names will likely be based on directory name.

**Comment by imthefrizzlefry** [comment]
I have been experimenting with this as well.  Recent updates to Immich may also ease some of your concerns.  Just a couple days ago, a new release of Immich started supporting XMP sidecar files.  I have been tempted to stand up a second instance of Immich that includes my digikam library as an "external library" with read only access.  In theory, that way I could view my library through Immich, while still managing and making changes through Digikam.

That said, I haven't set this up yet, but I think it would solve the issue I have now, which is 2 separate hard drives (one for Digikam and another for Immich.)  The one big downfall is that this flow wouldn't integrate well into the auto-upload flow I use for personal photos/videos I take with my phone.

**Comment by ronbruins** [comment]
I am using digikam and Immich and it works quite well. Immch just got the feature to read tags from exif data, however you need to rerun the extract metadata task (which goes for all assets) to get the metadata updated. or you can select the images and refresh metadata for a subset of them.

I have made a python script that creates albums based on the folder name, e.g.:

if a folder name ends with:

- @ it creates an album
- \# it adds it to a generic 'Year - Miscellaneous' album
- @ and contains specific keyword -> add to 'year - keyword' album

example (keyword is Horseriding):
2024-08-01 Holiday Spain @  (10 images)
2023-07-01 Holiday France @  (12 images)
2024-03-02 Baking # (3 images)
2024-02-01 Shopping # (4 images)
2024-01-04 Horseriding John @ (8 images)
2024-01-11 Horseriding John @ (6 images)


I get the following albums:
2024-08-01 Holiday Spain (10 images)
2023-07-01 Holiday France (12 images)
2024 Miscellaneous (7 images)
2024 Horseriding (14 images)

Now with the tagging ability I can make things even more granular to remove keywords from the title but use tags instead.

**Reply by kali-777** [reply]
I also looking for way to use DigiKam & some another SW preferable with some API to just view my Photos, but i do NOT want to duplicate Albums/photos/etc.. like Piwigo, so i just found Immich recently however its little bit "bummer"  Immich use PostgreSQL NOT mySQL/MariaDB/SQlite like almost every other SW?! 🤨 

@ronbruins This's script sounds amazing, I hope You're planning to publish! 🤩

I have NAS so my photos are there but i also have 3x RPIs for Home Assistant/HA & with 7" Display to replace my Google Nest Display, SO I'm also looking for way to integrate Immich with HA then display photos based on specific parameters, then I can even ask via voice HA just like a Google Home Assistant for example "show me pictures taken this location or with XY person or from last week, etc..." 😎

FYR 
[Immich HA Addon](https://github.com/outadoc/immich-home-assistant)
[Immich Kiosk & HA](https://github.com/damongolding/immich-kiosk)

**Reply by ronbruins** [reply]
I have enhanced the script and it now has the following functionality.

- create albums based on folders and/or tags:

  - Images in folders and tagged images are now automatically placed into albums. If the albums doesn't exist it is created, if it does exist it is added to that (shared) album.
  - To be able to have some granularity in creating albums I have the following suffixes on my folder / tags:
  - **@** (e.g. 2022-01-21 Holiday Spain @ )
  - **#** (e.g. 2022-03-18 sunset # ) 
  - @ - album will be created with the folder/tag name as album name and assets added. If album exists it will just add the assets
  - \# - generic album will be created e.g. 2022 Misc pictures and assets for that year will be added

- create consolidated albums based on GPS coordinates
  - in the settings specify the coordinates for a location:
  - `geo_dict['a specific location'] = {'lat': 51.323340222499525, 'lng': 3.8554779826176295}`
  - All images with GPS EXIF within a 300m radius will be added to album 'a specific location (radius is configurable)
 
- included a 'share veto' 
  - by default albums are shared with all users, in case this is not needed a share veto can be set to only share with specific users or not share at all.

- included a consolidated album option
  - if a keyword (specified in the settings) is part of the album/tag it will add the assets to a predefined album
  - e.g. `cons_albums['Soccer'] = "Soccer Album"`
    - This will add all assets in folder containing the word soccer into an album named 'Soccer Album'

I am in the process to clean the code up and put it in a public repo.

**Reply by ronbruins** [reply]
initial commit of my script https://github.com/ronbruins/immich-autoalbum I will work to get it commented, cleaner andbetter, as it was initially for my own use. It works for me :-)

**Comment by jucajuca** [comment]
There is this new package that allows you to work with sftp:

https://github.com/Demian98/immich-sftp-server

you can mount a remote sftp folder and store your digikam files in that folder.

**Comment by RobinEd** [comment]
I now find 4 solutions:
https://github.com/alvistar/immich-albums
https://github.com/Salvoxia/immich-folder-album-creator
https://github.com/ronbruins/immich-autoalbum
https://github.com/Demian98/immich-sftp-server

Just installed immich and am waiting for indexing. No idea which is best for my usecase (converting from Synology Photos and wanting to keep my folder based structure whilst sharing with my family)

This discussion also seems to be held over here: https://github.com/immich-app/immich/discussions/4279

**Comment by Bierchermuesli** [comment]
maybe a bit late the the party. Another approach would be a Digikam plugin which directly could push/sync with immich. if you have an opinion or idea how to implement please contribute on [Digikam Immich Wishlist](https://bugs.kde.org/show_bug.cgi?id=502592)
